src/tcpclientsrc.rs

```diff
@@ -378,6 +378,11 @@ impl TcpClientSrc {
                 events.push(event);
             }
         }
+
+        if buffer.get_size() == 0 {
+            events.push(gst::Event::new_eos().build());
```

Comment:
You probably also want to early-return here at some point and not push that 0-byte buffer downstream. But apart from that this looks all good

Reply:
So change that part and then squash the commits together as you suggested :)

Reply:
@sdroege There is the last one that needs to be corrected.

> As this is not a live source, reading should be started/stopped when switching between Ready and Paused. There should be no difference between Paused and Playing

I understand your comment, but I'm not sure how to change it.
Is it correct to call push_buffer function in Paused state?